src/NuGet.Core/NuGet.Packaging/Signing/TrustStore/X509TrustStore.cs

```diff
@@ -0,0 +1,143 @@
+            {
+                return Instance;
+            }
+
+            lock (LockObject)
+            {
+                if (Instance is not null)
+                {
+                    return Instance;
+                }
+
+                Instance = CreateX509ChainFactory(logger);
+            }
+
+            return Instance;
+        }
+
+        // Non-private for testing purposes only
+        internal static IX509ChainFactory CreateX509ChainFactory(ILogger logger)
+        {
+            if (IsEnabled)
+            {
+#if NET5_0_OR_GREATER
+                if (RuntimeEnvironmentHelper.IsLinux)
+                {
+                    if (SystemCertificateBundleX509ChainFactory.TryCreate(
+                        out SystemCertificateBundleX509ChainFactory systemBundleFactory))
+                    {
+                        logger.LogVerbose(
+                            string.Format(
+                                CultureInfo.CurrentCulture,
+                                Strings.ChainBuilding_UsingSystemCertificateBundle,
+                                systemBundleFactory.FilePath));
+
+                        return systemBundleFactory;
+                    }
+
+                    if (FallbackCertificateBundleX509ChainFactory.TryCreate(
+                        out FallbackCertificateBundleX509ChainFactory fallbackBundleFactory))
+                    {
+                        logger.LogVerbose(
+                            string.Format(
+                                CultureInfo.CurrentCulture,
+                                Strings.ChainBuilding_UsingFallbackCertificateBundle,
+                                fallbackBundleFactory.FilePath));
+
+                        return fallbackBundleFactory;
+                    }
+
+                    logger.LogVerbose(Strings.ChainBuilding_UsingNoCertificateBundle);
+
+                    return new NoCertificateBundleX509ChainFactory();
+                }
+
+                if (RuntimeEnvironmentHelper.IsMacOSX)
+                {
+                    if (FallbackCertificateBundleX509ChainFactory.TryCreate(
+                        out FallbackCertificateBundleX509ChainFactory fallbackBundleFactory))
+                    {
+                        logger.LogVerbose(
+                            string.Format(
+                                CultureInfo.CurrentCulture,
+                                Strings.ChainBuilding_UsingFallbackCertificateBundle,
+                                fallbackBundleFactory.FilePath));
+
+                        return fallbackBundleFactory;
+                    }
+
+                    logger.LogVerbose(Strings.ChainBuilding_UsingNoCertificateBundle);
+
+                    return new NoCertificateBundleX509ChainFactory();
+                }
+#endif
+            }
+
+            logger.LogVerbose(Strings.ChainBuilding_UsingDefaultTrustStore);
+
+            return new DefaultTrustStoreX509ChainFactory();
```

Comment:
>DefaultTrustStoreX509ChainFactory [](http://example.com/codeflow?start=23&length=33)

Also, do we want to allow customers to use the fallback trust store on Windows for verification or other purposes, for example?

Reply:
That's not a goal.

The fallback certificate bundle will ship in the .NET SDK, and there's technically nothing stopping someone from using it; however, it's out of scope to enable scenarios outside of this feature.